embassy-stm32/src/adc/v2.rs

```diff
@@ -167,6 +169,28 @@ where
         T::regs().cr2().modify(|reg| {
             reg.set_swstart(true);
         });
+    }
+
+    /// Perform a single conversion asynchronously.
+    async fn convert(&mut self) -> u16 {
+        self.start_conversion();
+
+        // wait for actual start and then finish
+        poll_fn(|_| {
```

Comment:
this should register the waker somewhere and wake it from the interrupt. If not, the task will sleep forever if Pending is returned.